### PR TITLE
Upgrade to JDK 21

### PR DESCRIPTION
**Pull request by lintool**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -224,7 +236,7 @@
               </transformers>
               <!-- This fixes the issue "Invalid signature file digest for Manifest main attributes"
                    cf. http://zhentao-li.blogspot.com/2012/06/maven-shade-plugin-invalid-signature.html -->
-              <filters>
+              <!--filters>
```
Comment (sahel-sh) [inline pom.xml:239]: is this change intentional?

**Review by sahel-sh — APPROVED** [review]
LGTM with a question

File: pom.xml (hunk 1 above)
Comment (lintool) [reply pom.xml:239]: Yes, commented out intentionally - not sure its effect... so punting for now, will circle back later to figure out implications.

**Review by MXueguang — COMMENTED** [review]
lgtm
